gateway-server/src/main/java/org/apache/knox/gateway/websockets/ProxyWebSocketAdapter.java

```diff
@@ -198,12 +232,29 @@ public void onError(Throwable cause) {
 
       @Override
       public void onMessageText(String message, Object session) {
+        LOG.logMessage("[From Backend <---]" + message);
+        remoteLock.lock();
         final RemoteEndpoint remote = getRemote();
```

Comment:
This should probably go inside of `try{` in case `getRemote()` throws an exception.